dts/arm/st/f3/stm32f3.dtsi

```diff
@@ -348,10 +348,10 @@
 			phase-seg2 = <6>;
 		};
 
-		adc1: adc@50000000 {
+		adc1: adc@40012400 {
 			compatible = "st,stm32-adc";
-			reg = <0x50000000 0x400>;
-			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x10000000>;
+			reg = <0x40012400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000200>;
```

Comment:
Actually, the initial code is (partially) right:
ADC1 is located at @5000000 for most variants of STM32F3 series. What's wrong is the bus identifier, should be STM32_CLOCK_BUS_AHB3.
If this address is valid for you, I guess you're using either stm32f373 or stm32f378 variant.

In that case, I'd advice the following:
Remove adc1 from stm32f3.dtsi and add it directly, with respective correct values in each of the SoC variant files: (stm32f302.dtsi, stm32f303.dtsi, ...)